packages/react-router-dom/index.tsx

```diff
@@ -471,8 +467,9 @@ export function RouterProvider({
   let setState = React.useCallback<RouterSubscriber>(
     (
       newState: RouterState,
-      { unstable_viewTransitionOpts: viewTransitionOpts }
+      { deletedFetchers, unstable_viewTransitionOpts: viewTransitionOpts }
     ) => {
+      deletedFetchers.forEach((key) => fetcherData.current.delete(key));
```

Comment:
The router tells us which fetchers can be deleted (once their ref counts go to 0)